src/components/util/TimestampInput.tsx

```diff
@@ -0,0 +1,135 @@
+/** *****************************************************************************
+ * Copyright 2020-2020 Exactpro (Exactpro Systems Limited)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ *  limitations under the License.
+ ***************************************************************************** */
+
+import * as React from 'react';
+import eventHttpApi from '../../api/event';
+import messageHttpApi from '../../api/message';
+import { formatTimestampValue } from '../../helpers/date';
+import { isEventAction } from '../../helpers/event';
+import { createBemElement } from '../../helpers/styleCreators';
+import { useDebouncedCallback } from '../../hooks';
+import { useOutsideClickListener } from '../../hooks/useOutsideClickListener';
+import { EventAction } from '../../models/EventAction';
+import { EventMessage } from '../../models/EventMessage';
+import { DateTimeMask } from '../../models/filter/FilterInputs';
+import Timestamp from '../../models/Timestamp';
+import KeyCodes from '../../util/KeyCodes';
+import '../../styles/timestamp-input.scss';
+
+const getTimestamp = (timestamp: Timestamp) => {
+	const ms = Math.floor(timestamp.nano / 1000000);
+	return +`${timestamp.epochSecond}${ms}`;
+};
+
+const dateFormatPattern = /[0-9]{4}-(0[1-9]|1[0-2])-(0[1-9]|[1-2][0-9]|3[0-1])/g;
+
+interface Props {
+	wrapperClassName?: string;
+	className?: string;
+	style?: React.CSSProperties;
+	readonly?: boolean;
+	placeholder?: string;
+	datepicker?: boolean;
+	onSubmit: (timestamp: number) => void;
+}
+
+const TimestampInput = (props: Props) => {
+	const {
+		wrapperClassName = 'timestamp-input',
+		datepicker = true,
+		readonly = false,
+		className = '',
+		placeholder = 'Go to ID or Timestamp',
+		style,
+		onSubmit,
+	} = props;
+
+	const [currentValue, setCurrentValue] = React.useState('');
+	const [currentTimestamp, setCurrentTimestamp] = React.useState(0);
+	const [showPicker, setShowPicker] = React.useState(false);
+
+	const ac = new AbortController();
```

Comment:
This should be inside of useEffect. You can pass abort signal to setTimestampFromEventOrMessage as argument

If one requests succeeds i think it's not a big deal to leave it hanging. It's only important we cancel previous requests on value change.